presidio-analyzer/presidio_analyzer/predefined_recognizers/in_vehicle_registration_recognizer.py

```diff
@@ -0,0 +1,395 @@
+from presidio_analyzer import Pattern, PatternRecognizer
+from typing import Optional, List, Tuple
+from presidio_analyzer.analyzer_utils import PresidioAnalyzerUtils as Utils
+
+
+class InVehicleRegistrationRecognizer(PatternRecognizer):
+    """
+    Recognizes Indian Vehicle Registration Number issued by RTO.
+
+    Reference(s):
+    https://en.wikipedia.org/wiki/Vehicle_registration_plates_of_India
+    https://en.wikipedia.org/wiki/Regional_Transport_Office
+    https://en.wikipedia.org/wiki/List_of_Regional_Transport_Office_districts_in_India
+
+    The registration scheme changed over time with multiple formats
+    in play over the years
+    India has multiple active patterns for registration plates issued to different
+    vehicle categories
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+        for different strings to be used during pattern matching.
+    This can allow a greater variety in input e.g. by removing dashes or spaces
+    """
+
+    PATTERNS = [
+        Pattern(
+            "India Vehicle Registration (Very Weak)",
+            r"\b[A-Z]{1}(?!0000)[0-9]{4}\b",
+            0.01,
+        ),
+        Pattern(
+            "India Vehicle Registration (Very Weak)",
+            r"\b[A-Z]{2}(?!0000)\d{4}\b",
+            0.01,
+        ),
+        Pattern(
+            "India Vehicle Registration (Very Weak)",
+            r"\b(I)(?!00000)\d{5}\b",
+            0.01,
+        ),
+        Pattern(
+            "India Vehicle Registration (Weak)",
+            r"\b[A-Z]{3}(?!0000)\d{4}\b",
+            0.2,
+        ),
+        Pattern(
+            "India Vehicle Registration (Medium)",
+            r"\b\d{1,3}(CD|CC|UN)[1-9]{1}[0-9]{1,3}\b",
+            0.40,
+        ),
+        Pattern(
+            "India Vehicle Registration",
+            r"\b[A-Z]{2}\d{1,2}[A-Z]{1,2}(?!0000)\d{4}\b",
+            0.50,
+        ),
+        Pattern(
+            "India Vehicle Registration",
+            r"\b[2-9]{1}[1-9]{1}(BH)(?!0000)\d{4}[A-HJ-NP-Z]{2}\b",
+            0.85,
+        ),
+        Pattern(
+            "India Vehicle Registration",
+            r"\b(?!00)\d{2}(A|B|C|D|E|F|H|K|P|R|X)\d{6}[A-Z]{1}\b",
+            0.85,
+        ),
+    ]
+
+    CONTEXT = ["RTO", "vehicle", "plate", "registration"]
+
+    # fmt: off
+    in_vehicle_foreign_mission_codes = [
```

Comment:
Thinking about performance, perhaps it would be better to have those as sets instead of lists. Sets lookup is O(1) while for lists it's O(n)

Reply:
agreed. it will save time especially for lists > 50 records for sure despite being ordered in values. incorporated.